src/grilodatasource.cpp

```diff
@@ -428,6 +437,7 @@ void GriloDataSource::grilo_source_result_cb(GrlSource *source, guint op_id,
             }
         }
         that->setFetching(false);
+        that->d->m_previouslyAddedId.clear();
```

Comment:
Just wondering if also error/cancellation situations handled earlier should also clear previous id?